client/app/bundles/common/DashboardPage.tsx

```diff
@@ -128,6 +129,20 @@ const DashboardPageRedirects = (): JSX.Element => {
 
   if (!courses?.length) return <Navigate to="/courses" />;
 
+  if (getUrlParameter('from') === 'auth') {
+    const visitedCourses = courses.filter((c) => !!c.lastActiveAt);
+
+    if (visitedCourses.length > 0) {
+      const latestSeenCourse = courses.reduce((c1, c2) =>
```

Comment:
since you are using `visitedCourses`, then you should rename the variable to `lastVisitedCourse`

Also you should perform reduce on `visitedCourses` instead of `courses`

Reply:
fixed